From 5c90b0b348f26f60d70a0379d538f0fc9b8bbfa7 Mon Sep 17 00:00:00 2001
From: Axel Huebl <axel.huebl@plasma.ninja>
Date: Thu, 11 Sep 2025 23:29:50 -0700
Subject: [PATCH] Pre-Build AMReX: RCylindrical, RSphere

Support an installed (1D) AMReX for the new RCylindrical, RSphere
modes. This is the typical (clean) install we do in package
managers like Conda and Spack.

Signed-off-by: Axel Huebl <axel.huebl@plasma.ninja>
---
 cmake/dependencies/AMReX.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmake/dependencies/AMReX.cmake b/cmake/dependencies/AMReX.cmake
index 6d5211c4b0..85825c5cdb 100644
--- a/cmake/dependencies/AMReX.cmake
+++ b/cmake/dependencies/AMReX.cmake
@@ -263,8 +263,10 @@ macro(find_amrex)
             set(COMPONENT_SIMD)
         endif()
 
-        set(WarpX_amrex_dim ${WarpX_DIMS})  # RZ is AMReX 2D
+        set(WarpX_amrex_dim ${WarpX_DIMS})
         list(TRANSFORM WarpX_amrex_dim REPLACE RZ 2)
+        list(TRANSFORM WarpX_amrex_dim REPLACE RCYLINDER 1)
+        list(TRANSFORM WarpX_amrex_dim REPLACE RSPHERE 1)
         list(REMOVE_DUPLICATES WarpX_amrex_dim)
         set(COMPONENT_DIMS)
         foreach(D IN LISTS WarpX_amrex_dim)
-- 
2.43.0

